src/notebooksyncagent.cpp

```diff
@@ -761,6 +762,16 @@ bool NotebookSyncAgent::calculateDelta(
     // separate them into buckets.
     // note that each remote URI can be associated with multiple local incidences (due recurrenceId incidences)
     // Here we can determine local additions and remote deletions.
+    KCalCore::Incidence::List additions, additionsSyncDate, addedPersistentExceptionOccurrences;
+    if (!mStorage->insertedIncidences(&additions, fromDate, mNotebook->uid()) ||
+        !mStorage->insertedIncidences(&additionsSyncDate, syncDateTime, mNotebook->uid())) {
+        LOG_CRITICAL("mKCal::ExtendedStorage::insertedIncidences() failed");
+        return false;
+    }
+    uniteIncidenceLists(additionsSyncDate, &additions);
```

Comment:
You're absolutely right.  I'll fix this (and the other cases where I did similar).